packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -65,6 +68,43 @@ function parseQueryParamsForScopedOption(id: string): scopedOption {
     };
 }
 
+function transformWarningToRollupWarning(
+    warning: CompilerDiagnostic,
+    src: string,
+    id: string
+): RollupWarning {
+    // For reference on RollupWarnings, a good example is:
+    // https://github.com/rollup/plugins/blob/53776ee/packages/typescript/src/diagnostics/toWarning.ts
+    const pluginCode = `LWC${warning.code}`; // modeled after TypeScript, e.g. TS5055
+    const result: RollupWarning = {
+        // Replace any newlines in case they exist, just so the Rollup output looks a bit cleaner
+        message: `@lwc/rollup-plugin: ${warning.message?.replace(/\n/g, ' ')}`,
+        plugin: PLUGIN_NAME,
+        pluginCode,
+    };
+    const { location } = warning;
+    if (location) {
+        result.loc = {
+            // The CompilerDiagnostic from @lwc/template-compiler reports an undefined filename, because it loses the
+            // filename context here:
+            // https://github.com/salesforce/lwc/blob/e2bc36f/packages/%40lwc/compiler/src/transformers/template.ts#L35-L38
+            file: id,
```

Comment:
It would be a nice improvement to pass the file name to the template compiler to produce more meaningful diagnostics.

Reply:
Agreed, but I left it out of this PR. We can always do it in the future, and the test verifies that the `file` property is correct.